File: ioloolo/BOJ/SolvedAc Class/Class 2/2609.java
import java.io.*;
import java.math.BigInteger;
import java.util.*;

public class Main {
	static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
	static BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));

	static StringTokenizer tokenizer;

	static int A;
	static int B;

	public static void main(String[] args) throws IOException {
		input();
		solve();

		writer.close();
		reader.close();
	}

	static void input() throws IOException {
		tokenizer = new StringTokenizer(reader.readLine());

		A = Integer.parseInt(tokenizer.nextToken());
		B = Integer.parseInt(tokenizer.nextToken());
	}

	static void solve() throws IOException {
		int gcd = BigInteger.valueOf(A).gcd(BigInteger.valueOf(B)).intValue();
		int lcm = A*B / gcd;

		writer.write(Integer.toString(gcd));
		writer.newLine();
		writer.write(Integer.toString(lcm));
	}
}

File: yuminn-k/BOJ/2609.py
firstVal, secondVal = map(int,input().split())

# 최대공약수
# firstVal & secondVal의 최대 공약수는 secondVal & firstVal를 secondVal로 나눈 나머지의 최대 공약수
def gcd(firstVal, secondVal):
    while secondVal > 0:
        firstVal, secondVal = secondVal, firstVal % secondVal
    return firstVal

# 최소공배수 
# firstVal와 secondVal의 곱을 firstVal와 secondVal의 최대 공약수로 나눈 값
def lcm(firstVal, secondVal):
    return firstVal * secondVal // gcd(firstVal, secondVal)

print(gcd(firstVal, secondVal))
print(lcm(firstVal, secondVal))